flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/table/catalog/hive/HiveCatalog.java

```diff
@@ -1403,4 +1446,29 @@ public CatalogColumnStatistics getPartitionColumnStatistics(ObjectPath tablePath
 		}
 	}
 
+	private static boolean createObjectIsGeneric(Map<String, String> properties) {
+		// When creating an object, a hive object needs explicitly have a key is_generic = false
```

Comment:
How about name it `getObjectIsGenericDefaultTrue`

Reply:
It's not just about the default value. For example, when creating an object and the `is_generic` key is missing, we'll add `is_generic=false` to the properties. 
Since the logic is different based on whether we're creating or retrieving an object, I think it's better to reflect that in the method names.

Reply:
The implementation is not really `create` the object but only append a property there.

Reply:
Sorry to hit the "resolve" button by mistake...

You're right but my point is we should differentiate whether we're creating or getting an object here. E.g. `createObjectIsGeneric` should be used in "create" methods like `createTable` or `createDatabase`, while `getObjectIsGeneric` should be used in "get" methods like `getTable` or `getDatabase`. It's clearer if the method names carry such information.

Reply:
I think each method should take care itself, if it only appends a property item, how about name it "appendIsGenericIfNecessary" or something like that ?

Reply:
Having "create" or "get" in the method names are more friendly to the method caller. The caller doesn't need to know whether the default value should be true or false, or whether a default value should be appended if missing. Instead, the caller just need to choose one method according to whether it's creating or getting an object.

Perhaps we can name them as `isGenericForCreate` and `isGenericForGet`? So that it's clear the method doesn't really `create` an object, but just to decide whether an object is generic in different scenarios. Does that sound better to you?

Reply:
I am with isGenericForCreate and isGenericForGet. -1 for `createObjectIsGeneric` and getObjectIsGeneric

Reply:
I'm also fine with name `isGenericForCreate` and `isGenericForGet`.